core_types/src/ugly.rs

```diff
@@ -0,0 +1,35 @@
+//! Types to better represent ugly aspects of the code.
+//! The idea is to bring more visibility to brittle or leaky abstractions, and to
+//! make bad code less convenient to use, so we are more motivated to fix it.
+//! Anything which is implemented using one of these types should be refactored
+//! so that we no longer have to use that type.
+
+/// Represents a value which may not have been initialized.
+/// Isomorphic to `Option`, but semantically different.
+#[derive(Clone)]
+pub enum Initable<T> {
+    Init(T),
+    Uninit,
+}
+
+impl<T> Initable<T> {
+    pub fn to_option(val: Initable<T>) -> Option<T> {
```

Comment:
You can just "mount" `self` as in:
```suggestion
    pub fn into_option(self) -> Option<T> {
```